### ISSUE TITLE
34c3 Autocrypt session

### ISSUE THREAD

**Issue by lefherz** [autocrypt/autocrypt#304]
On 34c3 there will be an Autocrypt session: https://events.ccc.de/congress/2017/wiki/index.php/Session:Autocrypt

We will have 3 hours, and a quite nice room.

**Comment by hpk42** [comment]
@xeniax will also be there and help and i think @azul  -- one question is how we want to do it ... Note that the session description can be changed, it's a wiki. 

And @pbrunschwig could you summarize the current state of enigmail/autocrypt support? Is it possible to get non-hacky way to enable/disable autocrypt with a normal enigmail config option? This page (added by @r10s says one as to go to the advanced editor https://autocrypt.org/install-autocrypt-enigmail.html

**Comment by hpk42** [comment]
and @r10s : do you think you could get incoming enigmail Autocrypt setup messages to work? Not a big deal if not -- people can still setup delta, and then install enigmail to receive the setup message and sync their secrets.  

Also @r10s : can you make a new set of login/librechat stickers like we used them at the freiburg session in november? 

@valodim : is there  a chance people can activate K-9 autoscrypt support and if so, could you create a little page on autocrypt.org like we have for enigmail?

**Comment by pbrunschwig** [comment]
@hpk42: 
the way to enable Autocrypt in Enigmail is as @r10s described. I won't work on changing this until 34c3. Apart from this, here is what is currently working:
- processing of autocrypt headers
- automatic import key from latest autocrypt message if no key available
- sending of autocrypt headers
- per-account setting for prefer-encrypt
- creation and processing of ASM

Not (yet) implemented:
- First run setup wizard/dialog
- Key Gossip
- link between autocrypt keys without email address and an email address (I might work on this in before the congress).
- GUI setting for enabling/disabling Autocrypt per account.

**Comment by r10s** [comment]
@hpk42 
- yes i create a new set of stickers
- incoming setup message: propably they won't work on 34c3, i'll continue hacking on this these days, but after all experiences with netpgp i fear there will be unexpected problems (the ui for incoming setup messages is already working can can be shown)

**Comment by hpk42** [comment]
So here is my suggestion. It's a UX session and i propose to model it on the freiburg one with modifications. In freiburg we gave 5 minutes with greeting, a little contextualizing, no slides, and went directly in with getting everybody installed with Delta -- this time because of a likely more advanced audience i suggestd to also get several folks installed with enigmail and maybe also K-9 (could be a small group lead by @valodim, likely going to be a more bumpy, not sure).   In Freiburg it was just Patrick who mailed with Delta folks and we showed the screenshot of enigmail on the projector etc.  
Pretty sure we can get several people installed with enigmail -- we can even send them to the autocrypt
web page which explains how to. 

Discussions and explanations of Autocrypt (when does it encrypt etc.) were mostly intertwined
with usage.  This is what led Hanna there to say this "wow, didn't know you can use pgp without an hour of explanations...". This impression is something to aim for: **it's not neccessary
to explain Autocrypt first to make use of it**.

As to the account setup process, @r10s has prepared stickers with account names and passwords on a custom domain.  Alternatively, participants can use their own accounts. I expect a mix of this to happen like it did in Freiburg.  

@xeniax also wanted to prepare some usage scenarios (creating a group etc.) and maybe we can prepare 2-3 tasks in the (hopefully likely case) that the first hour succeeeds in getting everyone into basic usage. 

Alternatively or in addition, we can close with a few slides to tell what precisely Autocrypt is, roadmap, open questions, how to contribute etc.

makes sense?

**Comment by azul** [comment]
The overall approach makes a lot of sense to me. 

I think there's a few challanges:
* people arriving late when everyone is focussed on their own devices
* people without devices or who do not want to install stuff on their devices
* keeping tension / focus alive
* transfering the larger group intro into work sessions into small group discussion

There's quite a bit of interest after the heise article i think. Not sure how many people will find it in the program and show up though.

Maybe we can have a rough schedule on the wall with a pointer of where in the programm we are so people who arrive late know what's happening and what will come later. Here's a sketch:

* greeting (5 Min.)
* context (5 Min.)
* installing delta chat, enigmail, k9 mail (45 min.)
* try out / interop testing (45 min.)
* breakout sessions with different focusses:
  * user experience / reflecting on the usage
  * security properties / comparison to other approaches
  * ?

Hmm... well probably end up having a lot of things in parallel. Not sure if structuring a 3 hour session will work. Anyone who could bring markers and larger sheets of paper? If not i will try and organize some during the congress.

**Comment by hpk42** [comment]
i just tried delta/enigmail interaction out again. Two questions came up:

- @pbrunschwig is there a way to set "prefer-encrypted=mutual" from the enigmail side? 
  Maybe a quick config-editor key that people can switch? (we are going to install nightly)

- @r10s is there a way to get rid of Delta showing "file-1.rfc822-headers" extra messages? 
  It's fine if it's then just available via an experimental site download for those who don't yet
  have delta and want to play with it.

**Comment by hpk42** [comment]
@azul up until "try out / interop testing" i am with you but a little skeptical of the rest (breakout session/working-group setting).  I'd rather get a few people setup e.g. with Autocrypt, Delta, K9, Enigmail issue trackers and prepare a single issue they'd like to see fixed.   Have some folks subscribe to the mailing list etc.  I.e. i'd rather end with "how can you help, open questions and issues", i.e. rather opening up than closing down.  

Integrating folks and asking them for help can also happen intermittently when one of us tries to help someone with a problem like "Oh, that's a bug, annooying -- do you have a github account? ... great, could you open an issue on autocrypt/autocrypt ..."

**Comment by pbrunschwig** [comment]
> is there a way to set "prefer-encrypted=mutual" from the enigmail side?
Maybe a quick config-editor key that people can switch? (we are going to install nightly)

@hpk42 yes, that's in the GUI: in the Account Settings under OpenPGP Security there is an option at the bottom: "Prefer encrypted emails from the people you mail with"

**Comment by r10s** [comment]
@hpk42  the rfc822-headers-files come from Memoryhole used in Enigmail, AFAIK Memoryhole is enabled by default in some Enigmail nighties.

Delta Chat parses these headers and Memoryhole since 0.10.0 which is not on F-Droid unfortunately but can be downloaded at https://github.com/deltachat/deltachat-android/releases

Alternatively, Memoryhole in Enigmail can be disabled at "Enigmail Preferences / Basics / Display Expert Settings and Menus / Advanced / Encrypt subject by default"

**Comment by hpk42** [comment]
@r10s cool -- then let's get some people at least to download the delta release instead of the f-droid one.

**Comment by Valodim** [comment]
> it's not neccessary to explain Autocrypt first to make use of it

This is a really cool point to make. I was a little worried that people at congress who come to such a meeting do want to know some details, but telling them along the way instead of holding a lecture might be a good way to go. That said, I can't shake the feeling that if I was a visitor myself, I would be expecting more details on the tech in this context and be rather disappointed to find a marketing event instead.

K-9 is kind of ok for sending and receiving (non-gossiped) autocrypt in the alpha, but the setup workflow is beyond horrible. I've been working on a version of K-9 that can be used for demonstration, but can't say how presentable that's going to be.

**Comment by hpk42** [comment]
@valodim there is plenty of room to discuss details -- we are going to be a number of people, and folks can ask detailed technical questions but there is no need to make everyone listen to it -- so i imagine the first 1-2 hours to consist of autocrypt knowledgable folks walking around and getting everyone installed, participating ... at the freiburg session we also explained things in between to the whole group but i wouldn't start with a whole block of explanations ahead of trying things out

**Comment by hpk42** [comment]
how i usually do sessions is trying to find out at the beginning, what people are up to and dynamically adapt.  if there are some folks who want to discuss rather than do hands-on we do a spin-off, or schedule it for later.

**Comment by hpk42** [comment]
@valodim the important thing is that android folks can install k-9 alpha somehow (without compiling themselves) ... the enigmail setup process is also tricky, that's fine.  is there a way to turn k-9 into prefer-encrypt=mutual?

**Comment by azul** [comment]
@hpk42 I'm not sure it makes sense to try and turn everything into prefer-encrypt=mutual. While this has a nice show effect it also gloses over some of the difficulties in real life situations that people might encounter. A large part of our considerations were about multi device situations with non autocrypt devices. I would be careful not to go down the track of trying to demo all automatic encryption.

**Comment by Valodim** [comment]
K-9 does not have a prefer-encrypt=mutual setting in the alpha (that hopefully becomes stable one of these days). There's no real reason, I just didn't add a switch yet. It's trivial to do, but won't end up in a stable very soon.

**Comment by petmos** [comment]
@pbrunschwig I try to communicate with one email account used by delta.chat and one used by thunderbird with enigmail nigthly.
Thunderbird sends encrypted emails to delta.chat and can be decrypted by delta.chat.
But delta.chat doesn't encrypt because there is no (valid) atocrypt header sent be enignmail.

I checked the enigmail header and found always:
autoencrypt: addr=XXXXXXXXX@XXXXX.XXX; prefer-encrypt=mutual; .... where XXXXXXXXX@XXXXX.XXX is always my delta.chat address.
Should't there be an entry with my thunderbird/enigmail address like: addr=my.thunderbird.account@addr.ess; prefer-encrypt=mutual too?

When I sent a message from enigmail there is always the autocrypt header included which was sent by delta.chat. I would expect a header from my enigmail/thunderbird account.

"Prefer encrypted emails from the people you mail with" is checked, too.

**Comment by petmos** [comment]
Okay - more info.
I have three different accounts on my thunderbird. One of them is the same as I use in delta.chat.
No I saw that emails sent with thunderbird always have the one and the same autocrypt header - the one from my delta.chat account.
This means:
Sent from account 1 to account 2: autocrypt header from delta.chat  
Sent from account 2 to account 1: autocrypt header from delta.chat  
Sent from account 1 to delta.chat: autocrypt header from delta.chat

**Comment by pbrunschwig** [comment]
The from address in the autocrypt header is taken from the Identity (which is not identical to the account) that you chose at the time of sending.

**Comment by petmos** [comment]
Okay, but can I chose the right identy. When I send from thunderbird I have use account 1 or account 2.

**Comment by pbrunschwig** [comment]
@petmos I tried to reproduce this, but it works fine for me with 2 accounts in the same Thunderbird.

**Comment by petmos** [comment]
I received an email with your reply and there is again the autocrypt header containing my delta.chat address and it's key.
![autocrypt](https://user-images.githubusercontent.com/1385807/34387374-14da545a-eb2e-11e7-8688-3a45b763d4ac.png)
So every email I send or receive from every account I have in thunderbird contains the same autocrypt header:
addr=[my.delta.chat@ADDRESS]; prefer-encrypt=mutual; keydata= [Always the same key]

**Comment by petmos** [comment]
@pbrunschwig I checked all messages sent by thunderbird with a second mail client k9.
Thunderbird never sent an autocrypt header. It is an error in the presentation inside thunderbird.
This was the cause why I always saw the same header.
When restart thunderbird and opens an email not sent by delta.chat or sent by thunderbird the header has now autocrypt part.
But when I opens a message from delta.chat I see this autocrypt header in **all** messages, even sent or received.

**Comment by dkg** [comment]
What's the status with this bug report?  34c3 is over, can we close this ticket?

**Comment by azul** [comment]
@petmos, Maybe it's a similar issue to https://sourceforge.net/p/enigmail/bugs/728/ ?

I have not been able to reproduce this myself. (But i have only tried with an existing mail account and key so far).
@petmos could you report the bug at the bugtracker above? Or Maybe we wait to see if the first report included enough info to reproduce / debug this.

I'll close this issue as like dkg said... 34C3 is over.

**Comment by petmos** [comment]
@azul, I have to create a sourceforge account and then I will report there.

**Comment by hpk42** [comment]
A proper closure of the 34c3 event would be to send a summary mail to the list about it.  While i helped preparing and pre-discussing things, and was present partially, i do not feel i have a full view.  I think the actual session diverged quite a bit from the pre-discussions i participated in (for good or bad, i don't know -- let's learn from how to collaboratively do these events!).  

Could @valodim, @azul, or @xenia help?  I am tentatively opening this issue and edit the title to strike the "prepare" bit.

**Comment by hpk42** [comment]
now that @kermoshina and others reported on the mailing list it's time to close this 34c3 issue, thanks everybody! :)
